src/Microsoft.ApplicationInsights/Managed/Shared/Extensibility/Implementation/Tracing/HeartbeatProvider.cs

```diff
@@ -0,0 +1,346 @@
+            IDictionary<string, HeartbeatPropertyPayload> defaultProps = defaultPayload.GetPayloadProperties();
+
+            this.sdkPayloadItems.Clear();
+            foreach (var kvpProp in defaultProps)
+            {
+                this.AddHealthPropertyInternal(this.sdkPayloadItems, kvpProp.Key, kvpProp.Value.PayloadValue, kvpProp.Value.IsHealthy);
+            }                
+        }
+
+        private void AddPropertiesToHeartbeat(MetricTelemetry hbeat, ConcurrentDictionary<string, HeartbeatPropertyPayload> props)
+        {
+            string updatedKeys = string.Empty;
+            string comma = hbeat.Properties.Count <= 0 ? string.Empty : ",";
+
+            lock (props)
+            {
+                foreach (var payloadItem in props)
+                {
+                    try
+                    {
+                        hbeat.Properties.Add(payloadItem.Key, payloadItem.Value.PayloadValue);
+                        hbeat.Sum += payloadItem.Value.IsHealthy ? 0 : 1;
+                        if (payloadItem.Value.IsUpdated)
+                        {
+                            string.Concat(updatedKeys, comma, payloadItem.Key);
+                            comma = ",";
+                        }
+                    }
+                    catch (Exception)
+                    {
+                        // skip sending this payload item out, no need to interrupt other payloads. Log it to core.
+                        CoreEventSource.Log.LogError(string.Format(CultureInfo.CurrentCulture, "Failed to send payload for item {0}.", payloadItem.Key));
+                    }
+                }
+            }
+
+            // update the special 'updated keys' property with the names of keys that have been updated.
+            if (!string.IsNullOrEmpty(updatedKeys))
+            {
+                hbeat.Properties[HeartbeatDefaultPayload.UpdatedFieldsPropertyKey] = updatedKeys;
+            }
+        }
+
+        private bool SetHealthPropertyInternal(ConcurrentDictionary<string, HeartbeatPropertyPayload> properties, string name, string payloadValue, bool? isHealthy)
+        {
+            try
+            {
+                properties.AddOrUpdate(name, (key) => 
+                {
+                    throw new Exception("Not allowed to set this!");
+                }, 
+                (key, property) =>
+                {
+                    if (isHealthy != null)
+                    {
+                        property.IsHealthy = isHealthy.Value;
+                    }
+                    if (payloadValue != null)
+                    {
+                        property.PayloadValue = payloadValue;
+                    }
+                    
+                    return property;
+                });
+            }
+            catch (Exception)
+            {
+                // swallow the exception and return false.
```

Comment:
why swallowing ex here?

Reply:
No good reason, likely leftover from a prior revision, removed it. We capture any thrown exception and log it in the public API method instead.